### PR TITLE
Travis build docs

### LINKED ISSUE #681: Let travis CI build and error-check the documentation
Currently, there is nothing but careful manual inspection stopping errors in the docs from being merged. This has led to problems: #665, #678, #212 (and there are probably more). The natural solution is to have travis CI check that the docs build cleanly.

Some challenges to keep in mind while working on this:

* sphinx can accept quite a lot of errors and still exit with a zero exit code: a decision on "how bad" things should be before we consider the doc build "failed" must be made, and then implemented
* if one of the examples fail to run under sphinx, it will be silently dropped from the example gallery (and some special circumstances can cause examples to fail under sphinx while working fine when run idividually, related to `__future__` imports)

The docs do not yet build on python3: #651. Still, error checking of the docs would be useful to have in place for python2 alonem which should be the first goal.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #681.  Boom.

Also fixed a problem where the `pil_plugin` overrides `imshow`.  Since we require matplotlib now, I removed `imshow` from the `pil_plugin`.

Could not build on 3.2 due to `u''` literals.  Could not build on 2.6 due to missing `flags` kwarg in `re.sub`.

### CODE REVIEW

File: skimage/filters/thresholding.py [hunk 1]
```diff
@@ -200,8 +200,10 @@ def threshold_isodata(image, nbins=256, return_all=False):
 
     Histogram-based threshold, known as Ridler-Calvard method or inter-means.
     Threshold values returned satisfy the following equality:
```
Comment (ahojnnes) [inline skimage/filters/thresholding.py]: `::` instead of `:` should also work here?

Comment (blink1073) [inline skimage/filters/thresholding.py]: Yep, I'll try that.

**Comment by ahojnnes** [comment]
Let me know when you are done here. Looks good to me.

**Comment by blink1073** [comment]
Go for it @ahojnnes!

File: skimage/filters/thresholding.py (hunk 1 above)
Comment (ahojnnes) [inline skimage/filters/thresholding.py]: For block syntax, you need to indent the content of the block. That's probably why it did not work?

Comment (blink1073) [inline skimage/filters/thresholding.py]: Perhaps, it renders nicely as-is though.

**Comment by stefanv** [comment]
@blink1073 Can we not raise an error for warnings on the doc build?  That would be ideal.

**Comment by blink1073** [comment]
@stefanv, that is the current behaviour. :smile:

**Comment by blink1073** [comment]
@stefanv, unless you meant you wanted to fail on Warnings, in which case please see the PR description.  :(

**Comment by stefanv** [comment]
Yes, the problem is that people write badly formatted docstrings that render with warnings, instead of cleanly--we need to be able to catch that.  If it is only a matter of the current warnings stopping you, I can get rid of them tonight.

**Comment by blink1073** [comment]
If you can puzzle out the solutions, go for it.

**Comment by blink1073** [comment]
FYI @stefanv , I'm poking at this some more...

**Comment by stefanv** [comment]
:+1:

**Comment by blink1073** [comment]
I'm stumped, I still could not puzzle out either of the two warnings I mentioned, and there are still 62 total warnings.

**Comment by stefanv** [comment]
``assert nr_of_warnings(output) <= 62``  :)

**Comment by blink1073** [comment]
The `toctree references unknown document` errors should have been taken care of by `numpydoc_show_class_members = False`, http://stackoverflow.com/questions/12206334/sphinx-autosummary-toctree-contains-reference-to-nonexisting-document-warnings, but they persist.

**Comment by blink1073** [comment]
Sleuthed it out: `numpydoc_class_members_toctree = False` does the trick, down to 34 warnings.

**Comment by stefanv** [comment]
That's progress, for sure!

**Comment by blink1073** [comment]
Ugh, that's enough for tonight...

**Comment by blink1073** [comment]
@stefanv, down to 23....

**Comment by blink1073** [comment]
...and 21, that's enough for today.

**Comment by stefanv** [comment]
Not a sprint, but a marathon!

**Comment by blink1073** [comment]
...and then there were 8.

**Comment by blink1073** [comment]
Well, it was 8 until I absorbed 8 more warnings from @jni's new and "improved" docs that should never be checked for warnings :-P.  (Back down to 8).

**Comment by blink1073** [comment]
Woohoo!  Down to 0 (locally)!  Where my golden plunger at?!?!?!

**Comment by blink1073** [comment]
I'll write up some notes for posterity to make this easier in the future.

**Comment by JDWarner** [comment]
Amazing work, @blink1073!

File: doc/README.md [hunk 2]
```diff
@@ -3,3 +3,16 @@ To build docs, run `make` in this directory. `make help` lists all targets.
...
 ## Requirements ##
 Sphinx is needed to build doc. Install with `pip install sphinx`.
+
+## Fixing Warnings ##
+
+- "citation not found: R###"
+  $ cd doc/build; grep -rn R### .
+  There is probably an underscore after the reference (e.g. [1]_)
+
+- "Duplicate ciation R###, other instance in...""
```
Comment (ahojnnes) [inline doc/README.md]: citation

**Comment by blink1073** [comment]
Done!  (Its Peanut Butter Jelly Time)[www.youtube.com/watch?v=LVPNXsc4wsQ].

**Comment by stefanv** [comment]
A documentation tour de force!  Congratulations and profound thanks, @blink1073 

Would you mind if I rebased this into a more compact form?  It would also give me the opportunity to sanitize the next-level comment war between you and @jni :)

**Comment by blink1073** [comment]
Go for it.

**Comment by stefanv** [comment]
Is one patch OK, or is there some logical way you'd like this split?

**Comment by blink1073** [comment]
It's a bit of a mess, really, I'd say one lump is as good as anything.

**Comment by blink1073** [comment]
I don't mind rebasing on my end.

**Comment by blink1073** [comment]
And done.

**Comment by stefanv** [comment]
And bam!

**Comment by stefanv** [comment]
Next question: how hard would it be to automatically update the website each time we merge something into master?  Would that be desirable?

**Comment by jni** [comment]
@blink1073 Hilarious! Anyone care to tell me what warnings I added so that I don't add them again in the future? =)

**Comment by blink1073** [comment]
@jni, your tables were nit indented and you had a character creep past the divider line.

**Comment by blink1073** [comment]
@stefanv, there is another PR that does just that. (I'm on my phone).

**Comment by jni** [comment]
@stefanv I think it's a great idea, @blink1073 awesome!

**Pull request merged**